toucan_connectors/toucan_connector.py

```diff
@@ -472,6 +472,10 @@ def get_identifier(self):
     def describe(self, data_source: ToucanDataSource):
         """ """
 
+    def get_target_version(self) -> str | None:
+        """Tries to get the version of the target SGBD/source/service"""
+        return None
```

Comment:
I'd rather have a mixin for connectors that support retrieving the version of their target, such as `DiscoverableConnector`. That way, the signature could be changed to `Self -> str`, and we would not have to call the method to know if a connector actually supports it

Reply:
Am not seeing how this can be done without calling a method 🤔, do you have an example please ?
Am okay with a class that need to be created and used just like DiscoverableConnector and i did that here : [`d0a9f06` (#662)](https://github.com/ToucanToco/toucan-connectors/pull/662/commits/d0a9f06262dc83541b99c93662e1f4bc7104faaf)

Reply:
`isinstance(conn, VersionableEngineConnector)` rather than `if (version := conn.get_engine_version()) is not None`